cmd/util/ledger/migrations/contract_cleanup_migration.go

```diff
@@ -0,0 +1,180 @@
+package migrations
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"sort"
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/cadence/runtime/common"
+
+	"github.com/onflow/flow-go/cmd/util/ledger/util/registers"
+	"github.com/onflow/flow-go/fvm/environment"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// ContractCleanupMigration normalized account's contract names and removes empty contracts.
+type ContractCleanupMigration struct {
+	log zerolog.Logger
+}
+
+var _ AccountBasedMigration = &ContractCleanupMigration{}
+
+func NewContractCleanupMigration() *ContractCleanupMigration {
+	return &ContractCleanupMigration{}
+}
+
+func (d *ContractCleanupMigration) InitMigration(
+	log zerolog.Logger,
+	_ *registers.ByAccount,
+	_ int,
+) error {
+	d.log = log.
+		With().
+		Str("migration", "ContractCleanupMigration").
+		Logger()
+
+	return nil
+}
+
+func (d *ContractCleanupMigration) MigrateAccount(
+	_ context.Context,
+	address common.Address,
+	accountRegisters *registers.AccountRegisters,
+) error {
+
+	// Normalize the account's contract names.
+	// This will deduplicate and sort the contract names.
+
+	contractNames, err := d.normalizeContractNames(accountRegisters)
+	if err != nil {
+		return fmt.Errorf(
+			"failed to normalize contract names for %s: %w",
+			address.HexWithPrefix(),
+			err,
+		)
+	}
+
+	// Cleanup the code for each contract.
+	// If the contract code is empty, it will be removed.
+
+	for _, contractName := range contractNames {
+		err = d.cleanupContractCode(accountRegisters, contractName)
+		if err != nil {
+			return fmt.Errorf(
+				"failed to cleanup contract code for %s: %w",
+				address.HexWithPrefix(),
+				err,
+			)
+		}
+	}
+
+	return nil
+}
+
+// normalizeContractNames deduplicates and sorts the account's contract names.
+func (d *ContractCleanupMigration) normalizeContractNames(
+	accountRegisters *registers.AccountRegisters,
+) (
+	[]string,
+	error,
+) {
+	owner := accountRegisters.Owner()
+
+	encodedContractNames, err := accountRegisters.Get(owner, flow.ContractNamesKey)
+	if err != nil {
+		return nil, fmt.Errorf(
+			"failed to get contract names: %w",
+			err,
+		)
+	}
+
+	if len(encodedContractNames) == 0 {
+		return nil, nil
+	}
```

Comment:
Maybe remove empty contract name register if   
`(encodedContractNames != nil) && len(encodedContractNames) == 0`.

Reply:
Good idea! Done in [0dbc20e](https://github.com/onflow/flow-go/pull/6148/commits/0dbc20e60ea3177b14b745bb63dd6fb2d773e90c)